visualization/app/codeCharta/ui/filePanel/filePanel.po.ts

```diff
@@ -4,16 +4,32 @@ export class FilePanelPageObject {
 		return page.$eval("file-panel-component md-select .md-text", element => element["innerText"])
 	}
 
+	async getSelectedChangedName(oldName: string) {
+		await page.waitForSelector("file-panel-component md-select .md-text")
+		await this.waitUntilNameChange(oldName)
+		return page.$eval("file-panel-component md-select .md-text", element => element["innerText"])
+	}
+
+	private async waitUntilNameChange(oldName: string) {
+		await page.waitForFunction(
```

Comment:
Should the listener really be a `waitForFunction`? I expect it to be triggered very frequently and that there's a convenient function that could be used more directly that does the same without triggering lots of executions?

Reply:
Dunno how to do it differently.
I guess there is a general problem whith dialog testing as angular, or material design theme (not sure about it since i'm new @it ) adds a couple of classes between the start and end of the animation loop.